Symfony mailer docs (#174)
mautic/user-documentation: docs/channels/emails.rst

Email delivery
##############

> Mautic delivers Emails using the method defined by the system administrator. If you are the system administrator for your Company, then you need to add the Email protocol for your Mautic instance to use. Mautic integrates with any Email service provider which offers SMTP mail servers. As Mautic uses the :xref:`Symfony Mailer` library since v5, it supports its core plugins. Transports for other email services might be found on Github or Packagist.

Reviewer: Vale fix and making it clearer about the core plugins.

```suggestion
As Mautic uses the :xref:`Symfony Mailer` library since v5, it supports all Symfony Mailer core plugins out of the box. Transports for other Email services might be found on GitHub or Packagist.
```

> SMTP Transport

Reviewer: Vale fix

```suggestion
SMTP transport
```

**************

> The SMTP transport is the default transport for Mautic. It's configured in the Mautic configuration under the Email Settings tab. The configuration is the same as in the :xref:`Symfony Mailer` documentation.

Reviewer: Making it a bit clearer for the end-user.

```suggestion
The SMTP transport is the default transport used for sending Emails with Mautic. It's configured in the Mautic configuration under the Email Settings tab. The configuration is the same as in the :xref:`Symfony Mailer` documentation.
```

> This is the example DSN mentioned in the :xref:`Symfony Mailer` documentation for SMTP:

Reviewer: First time we use a new abbreviation, we should use the full-text.

```suggestion
This is the example Delivery Status Notification (DSN) configuration mentioned in the :xref:`Symfony Mailer` documentation for SMTP:
```

.. code-block:: shell
    
    smtp://user:pass@smtp.example.com:port

> This is how this DSN converted to Mautic's email configuration:

Reviewer: I think this is what you mean?

```suggestion
Mautic creates this automatically from the values entered in the Email configuration:
```

@escopecz please add a table which explains what each of the values actually means. Should just need three columns I guess or maybe two - value name, what it means, any notes / links / references.  Docs on how to do the markdown are here: https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/lists_tables.html#tables. Thanks!

Author: Will do

Author: The table was added in https://github.com/mautic/user-documentation/pull/174/commits/7411e6172d793b1b2b1345ddfaf6192b8248b2e5

.. image:: images/emails/smtp-dsn.png
    :width: 400
    :alt: SMTP API DSN example

.. note::

>   Use the Mautic's global configuration to paste in the DSN information. Especially the API keys and passwords. The values must be URL-encoded and the configuration form will do that for you. If you are pasting DSN directly into the config/local.php file, you must URL-encode the values yourself.

Reviewer: Vale fix.

```suggestion
  Use the Mautic's global configuration to paste in the DSN information, especially the API keys and passwords. The values must be URL-encoded, and the configuration form does that for you. If you are pasting DSN directly into the config/local.php file, you must URL-encode the values yourself.
```

> Example API Transport Installation

Reviewer: ```suggestion

.. vale off

Example API transport installation

.. vale on

```

**********************************

> Let's say you want to use :xref:`Sendgrid` API instead of SMTP to send Emails. You can install the official Symfony Sendgrid Transport by running the following command that is mentioned along others in the :xref:`Symfony Mailer` documentation.

Reviewer: Vale fix

```suggestion
If you want to use :xref:`Sendgrid` API instead of SMTP to send Emails, for example, you can install the official Symfony Sendgrid Transport by running the following command that is mentioned along others in the :xref:`Symfony Mailer` documentation.
```

.. code-block:: shell
    

>     composer require symfony/sendgrid-mailer

Reviewer: What about people who are not using Composer? How do they use it?

Author: I don't think it makes sense to use those without Composer. Non-Composer Mautic installations will have to go for a Mautic plugin that does not have any other dependencies if SMTP is not enough.

Reviewer: I think that we need to make this clear up front then, that you need to be using composer? Or will be be supplying such plugins?

Author: I added a warning in https://github.com/mautic/user-documentation/pull/174/commits/e1850808dc6e8362c6338ec051280e2dd80c8e50

> After that, you can configure the transport in the Mautic configuration. The example DSN is again mentioned in the :xref:`Symfony Mailer` documentation along with other transports. In case of Sendgrid API, the DSN would look like this:

Reviewer: ```suggestion
After that, you can configure the transport in the Mautic configuration. The example DSN is again mentioned in the :xref:`Symfony Mailer` documentation along with other transports. In the example of using the Sendgrid API, the DSN looks like this:
```

.. code-block:: shell
    
    sendgrid+api://KEY@default

> This is how this DSN converted to Mautic's email configuration:

Reviewer: ```suggestion
This is how it would be set up in Mautic's Email configuration:
```

  .. image:: images/emails/sendgrid-api-dsn.png
    :width: 400
    :alt: Sendgrid API DSN example

Once you replace the Sendgrid API KEY, you can add it to the DSN and save the configuration. Mautic will now use the Sendgrid API to send Emails. 

.. note::

>   It is nice perk that Mautic can use any Symfony Mailer transport. However, be aware that such transport do not support batch sending even via API. They only send one email per request opposed to a thousand emails per request as some special Mautic tansports can. It also does not support transport callback handling. If you plan to send bigger ammounts of emails please consider using email transports built specifically for such use. These plugins are available in :doc:`Mautic Marketplace </marketplace/marketplace.rst>`.

Reviewer: ```suggestion
  It's nice perk that Mautic can use any transport provided by Symfony Mailer. However, be aware that such transports (from Symfony) don't support batch sending, even via API. They only send one email per request, as opposed to a thousand emails per request as is the case with some Mautic transports, which can make them slow at scale. They also don't support transport callback handling used for bounce management and reply detection. If you plan to send larger volumes of Emails or need to use features which require callback handling, please consider using Email transports built specifically for such use. These plugins are available in the :doc:`Mautic Marketplace </marketplace/marketplace.rst>`.
```

> The system can either send Emails immediately or queue them for processing in batches by a :doc:`cron job </configuration/cron_jobs>`.

Reviewer 2: I think we should kill the immediate mode from the docs and ask users to use the queue mode all the time

Author: The immediate mode is using `sync://` messenger option, right? Or are you talking about something else?

Reviewer 2: Yes, that is right. I suggest we do not use sync at all

Author: I thin it's useful when someone is just testing Mautic out or have Mautic with a few dozens of contacts. @RCheesley WDYT?

Reviewer: I do think that immediate mode has a place for low-volume sending or when the mails really do need to go out immediately.

I'd only be in favour of killing it if we're absolutely sure that every single user of Mautic on any hosting environment is going to be capable and willing to use the queueing system, and we can document how to make that effectively send the emails as if they were immediate.

.. code-block:: shell
    

>     php /path/to/mautic/bin/console messenger:consume email_transport

Reviewer: Please update also the cron jobs file, and the command line interfaces file. Both will need to have this command included and explained.
